include/swift/Sema/ConstraintSystem.h

```diff
@@ -1547,6 +1575,13 @@ class Solution {
   llvm::MapVector<const CaseLabelItem *, CaseLabelItemInfo>
       caseLabelItems;
 
+  /// Maps catch nodes to the set of potential throw sites that will be caught
+  /// at that location.
+
+  /// The set of opened types for a given locator.
+  std::vector<std::pair<CatchNode, PotentialThrowSite>>
+      potentialThrowSites;
```

Comment:
Nit: please add the size to `Solution::getTotalMemory()`

Reply:
Sure, I'll do this when I rework the data structure here. Thank you!